tests/unit/backend/chalice/api_server/test_revisions.py

```diff
@@ -232,57 +231,82 @@ def test__delete_published_dataset_during_revision(self):
         resp = self.app.delete(test_dataset_url, headers=self.headers)
         resp.raise_for_status()
 
+        # Get the revision
+        resp = self.app.get(self.test_url_collect_private, headers=self.headers)
+        resp.raise_for_status()
+
         # The dataset is a tombstone in the revision
+        self.assertEqual(json.loads(resp.body)["datasets"], [])
         self.session.expire_all()
         for dataset in self.rev_collection.datasets:
             if dataset.id == rev_dataset_id:
                 self.assertTrue(self.rev_collection.datasets[0].tombstone)
                 break
+        self.assertPublishedCollectionOK(expected_body, pub_s3_objects)
 
-        # Get revision by owner, tombstone visible
-        resp = self.app.get(self.test_url_collect_private, headers=self.headers)
-        resp.raise_for_status()
-        self.assertEqual(json.loads(resp.body)["datasets"][0]["id"], rev_dataset_id)
 
-        # Get revision, tombstone invisible
-        resp = self.app.get(self.test_url_collect_private)
-        resp.raise_for_status()
-        self.assertEqual(json.loads(resp.body)["datasets"], [])
+class TestPublishRevision(BaseRevisionTest):
+    def setUp(self):
+        super().setUp()
+        self.test_url = f"/dp/v1/collections/{self.pub_collection.id}/publish"
 
-        # Published assets are not deleted
-        self.assertPublishedCollectionOK(expected_body, pub_s3_objects)
+    def publish_collection(self, collection_id: str):
```

Comment:
Would it be worth putting this function in `BaseRevisionTest` or even `BaseAuthAPITest`?

Reply:
We can move it once another test needs it.